spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/KafkaStreamingSource.java

```diff
@@ -0,0 +1,184 @@
+    this.conf = conf;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    super.configurePipeline(pipelineConfigurer);
+    // check the schema if there is one
+    if (!Strings.isNullOrEmpty(conf.schema)) {
+      if (Strings.isNullOrEmpty(conf.format)) {
+        throw new IllegalArgumentException("You must specify a format to go along with a schema.");
+      }
+      FormatSpecification formatSpec = conf.getFormatSpec();
+
+      try {
+        RecordFormat recordFormat = RecordFormats.createInitializedFormat(formatSpec);
+        pipelineConfigurer.getStageConfigurer().setOutputSchema(recordFormat.getSchema());
+      } catch (Exception e) {
+        throw new IllegalArgumentException("Invalid format and schema: " + e.getMessage(), e);
+      }
+    } else {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(DEFAULT_SCHEMA);
+    }
+  }
+
+  @Override
+  public JavaDStream<StructuredRecord> getStream(JavaStreamingContext javaStreamingContext) throws Exception {
+    Map<String, String> kafkaParams = new HashMap<>();
+    kafkaParams.put("metadata.broker.list", conf.brokers);
+
+    final FormatSpecification formatSpec = conf.getFormatSpec();
+    JavaPairInputDStream<byte[], byte[]> kafkaStream = KafkaUtils.createDirectStream(
+      javaStreamingContext, byte[].class, byte[].class, DefaultDecoder.class, DefaultDecoder.class,
+      kafkaParams, conf.getTopics());
+
+    if (formatSpec == null) {
+      return kafkaStream.map(new Function<Tuple2<byte[], byte[]>, StructuredRecord>() {
+        @Override
+        public StructuredRecord call(Tuple2<byte[], byte[]> in) throws Exception {
+          return StructuredRecord.builder(DEFAULT_SCHEMA)
+            .set("key", in._1())
+            .set("message", in._2())
+            .build();
+        }
+      });
+    }
+
+    return kafkaStream.map(new FormatFunction(conf.schema, conf.format));
+  }
+
+  private static class FormatFunction implements Function<Tuple2<byte[], byte[]>, StructuredRecord> {
+    private final String schemaStr;
+    private final String format;
+    private transient RecordFormat<StreamEvent, StructuredRecord> recordFormat;
+
+    public FormatFunction(String schemaStr, String format) {
+      this.schemaStr = schemaStr;
+      this.format = format;
+    }
+
+    @Override
+    public StructuredRecord call(Tuple2<byte[], byte[]> in) throws Exception {
+      if (recordFormat == null) {
+        Schema schema = Strings.isNullOrEmpty(schemaStr) ? null : Schema.parseJson(schemaStr);
+        FormatSpecification spec = new FormatSpecification(format, schema, new HashMap<String, String>());
+        recordFormat = RecordFormats.createInitializedFormat(spec);
+      }
+      return recordFormat.read(new StreamEvent(ByteBuffer.wrap(in._2())));
+    }
+  }
+
+  /**
+   * Conf for Kafka streaming source.
+   */
+  public static class Conf extends PluginConfig {
```

Comment:
in order to use it, StreamingSource needs some additional capabilities. Opened HYDRA-372.